testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -3322,6 +3323,99 @@ public void testCreateSchemaWithNonLowercaseOwnerName()
+    }
+
+    @Test
+    public void testCreateOrReplaceTableWhenTableAlreadyExistsSameSchema()
+    {
+        String table = "test_create_or_replace_" + randomNameSuffix();
+        @Language("SQL") String query = "SELECT nationkey, name, regionkey FROM nation";
+        @Language("SQL") String rowCountQuery = "SELECT count(*) FROM nation";
+        if (!hasBehavior(SUPPORTS_CREATE_OR_REPLACE_TABLE)) {
+            assertQueryFails("CREATE OR REPLACE TABLE " + table + " AS " + query, "This connector does not support replacing tables");
+            return;
+        }
+
+        assertUpdate("CREATE TABLE " + table + " AS SELECT CAST(1 AS BIGINT) AS nationkey, 'test' AS name, CAST(2 AS BIGINT) AS regionkey FROM nation LIMIT 1", 1L);
+        assertUpdate("CREATE OR REPLACE TABLE " + table + " AS " + query, rowCountQuery);
+        assertQuery("SELECT * FROM " + table, query);
+        assertUpdate("DROP TABLE " + table);
+
+        assertFalse(getQueryRunner().tableExists(getSession(), table));
+    }
+
+    @Test
+    public void testCreateOrReplaceTableWhenTableAlreadyExistsSameSchemaNoData()
+    {
+        String table = "test_create_or_replace_" + randomNameSuffix();
+        @Language("SQL") String query = "SELECT nationkey, name, regionkey FROM nation LIMIT 0";
+        if (!hasBehavior(SUPPORTS_CREATE_OR_REPLACE_TABLE)) {
+            assertQueryFails("CREATE OR REPLACE TABLE " + table + " AS " + query, "This connector does not support replacing tables");
+            return;
+        }
+
+        assertUpdate("CREATE TABLE " + table + " AS SELECT nationkey, name, regionkey FROM nation", "SELECT count(*) FROM nation");
+        assertUpdate("CREATE OR REPLACE TABLE " + table + " AS " + query, 0L);
+        assertQuery("SELECT * FROM " + table, query);
+        assertUpdate("DROP TABLE " + table);
+
+        assertFalse(getQueryRunner().tableExists(getSession(), table));
+    }
+
+    @Test
+    public void testCreateOrReplaceTableWhenTableAlreadyExistsDifferentSchemaNewColumnsDroppedColumns()
+    {
+        String table = "test_create_or_replace_" + randomNameSuffix();
+        @Language("SQL") String query = "SELECT nationkey AS nationkey_new, name AS name_new_2, regionkey AS region_key_new FROM nation";
+        @Language("SQL") String rowCountQuery = "SELECT count(*) FROM nation";
+        if (!hasBehavior(SUPPORTS_CREATE_OR_REPLACE_TABLE)) {
+            assertQueryFails("CREATE OR REPLACE TABLE " + table + " AS " + query, "This connector does not support replacing tables");
+            return;
+        }
+
+        assertUpdate("CREATE TABLE " + table + " AS SELECT nationkey, name, regionkey FROM nation", "SELECT count(*) FROM nation");
+        assertUpdate("CREATE OR REPLACE TABLE " + table + " AS " + query, rowCountQuery);
+        assertQuery("SELECT * FROM " + table, query);
+        assertUpdate("DROP TABLE " + table);
+
+        assertFalse(getQueryRunner().tableExists(getSession(), table));
+    }
+
```

Comment:
The PR advertises _atomic_ replacement of a table.

Let's add another test in BCT in a similar fashion as what has been done in `io.trino.testing.BaseConnectorTest#testAddColumnConcurrently` where a bunch of `SELECT` statements is mixed with a CORTAS statement in a concurrent fashion and all the `SELECT` statements should be successful.